packages/server/src/middleware/nuxt.js

```diff
@@ -7,7 +7,22 @@ import { getContext } from '@nuxt/utils'
 export default ({ options, nuxt, renderRoute, resources }) => async function nuxtMiddleware(req, res, next) {
   // Get context
   const context = getContext(req, res)
-  const url = decodeURI(req.url)
+  let url
+
+  try {
+    url = decodeURI(req.url)
+  } catch (err) {
+    /* istanbul ignore if */
+    if (context && context.redirected) {
+      consola.error(err)
+      return err
+    }
+
+    const e = new Error('URL malformed')
```

Comment:
I'd love to see a more descriptive variable name here :relaxed:

Reply:
Maybe you could use the error coming from the failed decoding directly anyway? :thinking: 

(see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURI#Catching_errors)

Reply:
Moved everything in the same `try` block and setting `statusCode = 400` in the `catch`.